tools/azure-devtools/src/azure_devtools/scenario_tests/preparers.py

```diff
@@ -126,6 +126,12 @@ def _preparer_wrapper(test_class_instance, **kwargs):
                         _logger.debug("Storing cached resource for %s", self.__class__.__name__)
                         AbstractPreparer._resource_cache[aggregate_cache_key] = AbstractPreparer.ResourceCacheEntry(resource_name, kwargs, self)
 
+            if test_class_instance.is_live:
```

Comment:
FYI i'm hoping to slip in https://github.com/Azure/azure-sdk-for-python/pull/11878 first so we get proper papertrail if someone git-blames this code (which may cause a conflict here).  Just calling this out so no surprises.  (that PR is ready and just waiting on a storage hotfix to pass tests)

Reply:
@yunhaoling did you change and push this file? This question was for you I guess.

Reply:
@KieranBrantnerMagee I merged your branch and as you've get #11878 merged and we don't get any conflict here. I think everything is fine.